tests/cpp_api/audio_tests/audio_tests.cpp

```diff
@@ -0,0 +1,208 @@
+    const char *rocal_data_path = std::getenv("ROCAL_DATA_PATH");
+    if (strcmp(rocal_data_path, "") == 0) {
+        std::cout << "\n ROCAL_DATA_PATH env variable has not been set. ";
+        exit(0);
+    }
+
+    std::string ref_file_path = std::string(rocal_data_path) + "rocal_data/GoldenOutputsTensor/reference_outputs_audio/" + case_name + "_output.bin";
+    long int out_buffer_size = frames;
+    std::vector<float> ref_output(out_buffer_size);
+    std::fstream fin(ref_file_path, std::ios::in | std::ios::binary);
+    if (fin.is_open()) {
+        for (long int i = 0; i < out_buffer_size; i++) {
+            if (!fin.eof()) {
+                fin.read(reinterpret_cast<char *>(ref_output.data()), sizeof(float));
+            } else {
+                std::cout << "\nUnable to read all data from golden outputs\n";
+                return pass_status;
+            }
+        }
+    } else {
+        std::cout << "\nCould not open the reference output. Please check the path specified\n";
+        return pass_status;
+    }
+
+    int matched_indices = 0;
+    for (int j = 0; j < frames; j++) {
+        float ref_val, out_val;
+        ref_val = ref_output[j];
+        out_val = dst_ptr[j];
+        bool invalid_comparison = ((out_val == 0.0f) && (ref_val != 0.0f));
+        if (!invalid_comparison && abs(out_val - ref_val) < 1e-20)
+            matched_indices += 1;
+    }
+
+    std::cout << std::endl << "Results for Test case: " << std::endl;
+    if ((matched_indices == frames) && matched_indices != 0) {
+        pass_status = true;
+    }
+
+    return pass_status;
+}
+
+int test(int test_case, const char *path, int qa_mode, int downmix, int gpu);
+int main(int argc, const char **argv) {
+    // check command-line usage
+    const int MIN_ARG_COUNT = 2;
+    if (argc < MIN_ARG_COUNT) {
+        printf("Usage: ./audio_tests <audio-dataset-folder> <test_case> <downmix=0/1> <device-gpu=1/cpu=0> <qa_mode=0/1>\n");
+        return -1;
+    }
+
+    int argIdx = 0;
+    const char *path = argv[++argIdx];
+    int qa_mode = 0;
+    unsigned test_case = 0;
+    bool downmix = false;
+    bool gpu = 0;
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        test_case = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        downmix = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        gpu = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        qa_mode = atoi(argv[++argIdx]);
+
+    if (gpu) {  // TODO - Will be removed when GPU support is added for Audio pipeline
+        std::cout << "WRN : Currently Audio unit test supports only HOST backend\n";
+        gpu = false;
+    }
+
+    int return_val = test(test_case, path, qa_mode, downmix, gpu);
+    return return_val;
+}
+
+int test(int test_case, const char *path, int qa_mode, int downmix, int gpu) {
+    int input_batch_size = 1;
+    bool is_output_audio_decoder = false;
+    std::cout << ">>> test case " << test_case << std::endl;
```

Comment:
remove all these extra `>>>`. I am cleaning up other tests also. Output is very hard to read